instrumentation-genai/opentelemetry-instrumentation-langchain/src/opentelemetry/instrumentation/langchain/span_manager.py

```diff
@@ -0,0 +1,115 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional
+from uuid import UUID
+
+from opentelemetry.context import Context, get_current
+from opentelemetry.semconv._incubating.attributes import (
+    gen_ai_attributes as GenAI,
+)
+from opentelemetry.semconv.attributes import (
+    error_attributes as ErrorAttributes,
+)
+from opentelemetry.trace import Span, SpanKind, Tracer, set_span_in_context
+from opentelemetry.trace.status import Status, StatusCode
+
+
+@dataclass
+class _SpanState:
+    span: Span
+    context: Context
+    children: List[UUID] = field(default_factory=list)
+
+
+class SpanManager:
+    def __init__(
+        self,
+        tracer: Tracer,
+    ) -> None:
+        self._tracer = tracer
+
+        # Map from run_id -> _SpanState, to keep track of spans and parent/child relationships
+        self.spans: Dict[UUID, _SpanState] = {}
+
+    def create_span(
+        self,
+        run_id: UUID,
+        parent_run_id: Optional[UUID],
+        span_name: str,
+        kind: SpanKind = SpanKind.INTERNAL,
+    ) -> Span:
+        if parent_run_id is not None and parent_run_id in self.spans:
+            parent_span = self.spans[parent_run_id].span
+            ctx = set_span_in_context(parent_span)
+            span = self._tracer.start_span(
+                name=span_name, kind=kind, context=ctx
+            )
+        else:
+            # top-level or missing parent
+            span = self._tracer.start_span(name=span_name, kind=kind)
+
+        span_state = _SpanState(span=span, context=get_current())
+        self.spans[run_id] = span_state
+
+        if parent_run_id is not None and parent_run_id in self.spans:
```

Comment:
Same condition as on line 54? Seems like the next line can be moved into that block.